checks/binary_artifact.go

```diff
@@ -26,7 +26,10 @@ const CheckBinaryArtifacts string = "Binary-Artifacts"
 
 //nolint
 func init() {
-	if err := registerCheck(CheckBinaryArtifacts, BinaryArtifacts); err != nil {
+	var supportedRequestTypes = []checker.RequestType{
+		checker.FileBased,
+	}
+	if err := registerCheck(CheckBinaryArtifacts, BinaryArtifacts, supportedRequestTypes); err != nil {
```

Comment:
fyi, when we added the `--local` repo, we decided to have the metadata in the checks.yaml rather than hardcoded in the implementation. What's the motivation for changing it?

Reply:
> we decided to have the metadata in the checks.yaml rather than hardcoded in the implementation.

True. After our current implementation I have found it confusing that the `docs` package validates the correctness of `checks`. My main motivation is changing this is so that each check is complete in itself and can self-validate, if that makes sense. 

Let me know. Will wait on your response to submit this PR.

Reply:
Moving here seems OK to me. Assuming I'm understanding this correctly, this level of granularity like an implementation detail that users don't really have to understand. 

`local` on the other hand would be useful for users of scorecards to know and be documented in `checks.yaml`/`checks.md`.

Reply:
Or actually maybe I've misunderstood this -- the plan is to move the source of truth for "local" completely to being in the implementation. That seems fine to me, and while our output `checks.md` doesn't show this today, in the future we might want to show this there as well. 

Either way, this seems fine to me.

Reply:
Thanks! I'm merging this in that case.